tools/wpt/browser.py

```diff
@@ -293,9 +293,17 @@ class Chrome(Browser):
     """
 
     product = "chrome"
-    binary = "/usr/bin/google-chrome"
     requirements = "requirements_chrome.txt"
 
+    @property
+    def binary(self):
+        if uname[0] == "Linux":
+            return "/usr/bin/google-chrome"
+        if uname[0] == "Darwin":
+            return "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
```

Comment:
Technically, I don't think this is guaranteed. I think technically on macOS we should read Contents/Info.plist, read the CFBundleExecutable key, and then set the binary to Contents/MacOS/[that value]. That said, probably needless pedantry.

Reply:
You're right; this is BAD(tm). Also the bundle might be installed to a user-specific applications directory. Not to mention I made zero effort for Windows...

I'm essentially using a simplified version of Chromedriver's heuristics: https://cs.chromium.org/chromium/src/chrome/test/chromedriver/chrome/chrome_finder.cc , which is also not ideal.

Basically the only reliable way is to always explicitly specify the --binary flag.

Reply:
I at some point had some plan to implement some `find_macos_binary_path(bundle_id)`, which took the application id (e.g., `com.google.Chrome`) and then returns the path to the binary. Probably even splitting it up into two functions (one which finds the `.app` bundle, one which finds the binary within it).

Reply:
Let's wish it'll happen some day :-P